python/beeai_framework/tools/tool.py

```diff
@@ -137,6 +196,17 @@ class FunctionTool(Tool):
         description = tool_description
         input_schema = tool_input
 
+        def __init__(self, options: dict[str, Any] | None = None) -> None:
+            super().__init__(options)
+            # replace any non-alphanumeric char with _
+            formatted_name = re.sub(r"\W+", "_", self.name).lower()
```

Comment:
I noticed that you were already using this helper somewhere, please create a utility function in `utils/strings.py`

Reply:
https://github.com/i-am-bee/beeai-framework/pull/429/commits/53b029bcc960660a29eb76c924a51e7356b6b644